service/src/main/java/uk/gov/hmcts/reform/fpl/config/feign/errordecoder/IdamErrorDecoder.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.fpl.config.feign.errordecoder;
+
+import feign.Response;
+import feign.RetryableException;
+import feign.codec.ErrorDecoder;
+import org.springframework.http.HttpStatus;
+
+public class IdamErrorDecoder implements ErrorDecoder {
+
+    private final ErrorDecoder defaultDecoder;
+
+    public IdamErrorDecoder() {
+        this(new Default());
+    }
+
+    public IdamErrorDecoder(ErrorDecoder defaultDecoder) {
+        this.defaultDecoder = defaultDecoder;
+    }
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        Exception exception = defaultDecoder.decode(methodKey, response);
+
+        if (exception instanceof RetryableException) {
+            return exception;
+        }
+
+        if (HttpStatus.valueOf(response.status()).is5xxServerError()) {
+            return new RetryableException(
```

Comment:
throwing RetrywableException can make change not transparent to existing code. If there is code that do try{}catch(SomeException ex), ex will be wrapped in RetryableException and catch wont be trigger.

there is option to configure propagation policy
    @Bean
    ExceptionPropagationPolicy propagationPolicy() {
        return ExceptionPropagationPolicy.UNWRAP;
    }
    
  Your change is about http 5xx family only so i guess it should be fine (dont expect to have catch(Feign.InternalServerError)), but worth checking.
  
In general it is preferred to make retries transparent so i would consider using ExceptionPropagationPolicy.UNWRAP

Reply:
Yeah, makes sense. Added